state/application.go

```diff
@@ -1094,6 +1112,8 @@ func (a *Application) SetCharm(cfg SetCharmConfig) (err error) {
 		return errors.Annotate(err, "validating config settings")
 	}
 
+	// TODO validate lxd profile?
```

Comment:
Potentially, but I'm not sure it's required? Maybe we can validate the profile but not return an error.

Reply:
I'm going to leave this as a to do, some related structures need force woven thru to properly accomplish this.